client/src/main/java/org/eclipse/hono/client/impl/HonoConnectionImpl.java

```diff
@@ -376,11 +366,7 @@ private void connect(
                         conAttempt -> {
                             connecting.compareAndSet(true, false);
                             if (conAttempt.failed()) {
-                                if ((!isIgnoreTerminalCondition()) && isTerminalConnectionError(conAttempt.cause())) {
-                                    failConnectionAttempt(conAttempt.cause(), connectionHandler);
-                                } else {
-                                    reconnect(conAttempt.cause(), connectionHandler, disconnectHandler);
-                                }
+                                reconnect(conAttempt.cause(), connectionHandler, disconnectHandler);
```

Comment:
IMHO we should still log out specific error messages for the former terminal types of connection errors as we do in the *failConnectionAttempt* method (which is now never invoked with the default reconnect attempts set to -1). Otherwise it might be hard for operators to fix the problem ...

Reply:
But that wouldn't be any "terminal errors" but more something like "noteworthy errors".

Reply:
I added the logging, based on the previous classification.